appserver/payara-appserver-modules/security-openid/src/main/java/fish/payara/security/openid/OpenIdAuthenticationMechanism.java

```diff
@@ -214,9 +220,30 @@ public AuthenticationStatus validateRequest(
                 throw new AuthenticationException("Failed to register CallerPrincipalCallback.", ex);
             }
 
-            if (configuration.isTokenAutoRefresh()) {
+            LogoutConfiguration logout = configuration.getLogoutConfiguration();
+            boolean accessTokenExpired = this.context.getAccessToken().isExpired();
+            boolean identityTokenExpired = this.context.getIdentityToken().isExpired();
+            if (logout.isIdentityTokenExpiry()) {
+                LOGGER.log(Level.FINE, "UserPrincipal is set, check if Identity Token is valid.");
+            }
+            if (logout.isAccessTokenExpiry()) {
                 LOGGER.log(Level.FINE, "UserPrincipal is set, check if Access Token is valid.");
-                return this.reAuthenticate(request, response, httpContext);
+            }
+
+            if((logout.isAccessTokenExpiry() && accessTokenExpired)
+                    || (logout.isIdentityTokenExpiry() && identityTokenExpired)) {
+                if (configuration.isTokenAutoRefresh()) {
```

Comment:
They way you've nested this means that `tokenAutoRefresh` is ignored if at least one of `accessTokenExpiry` or `identityTokenExpiry` isn't true. 

On one hand, I think this makes sense because if a user has set `tokenAutoRefresh` then logically the application requires a valid access token to function. On the other hand, maybe someone wants to automatically refresh tokens, but they don't want the user to be logged-out if the tokens expire. 

If this _is_ the intended behaviour, then the documentation should be clear that `tokenAutoRefresh` only works when at least one of `accessTokenExpiry` or `identityTokenExpiry` is set.

Reply:
Hi,

Yes, it is intended behavior, I will update the documentation. Also IMO `accessTokenExpiry` should be true by default, WDYT?

Reply:
Access tokens are usually short-lived. It's 5 minutes by default on Keycloak. So if `accessTokenExpiry` is true by default, then maybe `tokenAutoRefresh` should also be true by default otherwise that's a really short idle timeout (assuming 5 minutes). See what I mean? 

I think you can make a good case for either being the default, true or false. On one hand `false` is simpler and reduces network traffic, on the other hand I think most people will want to use `true` once they realize the benefits of tying their Payara session to OIDC token expiry.